lib/src/command_runner/completion_command_runner.dart

```diff
@@ -71,10 +71,18 @@ abstract class CompletionCommandRunner<T> extends CommandRunner<T> {
     final reservedCommands = [
       HandleCompletionRequestCommand.commandName,
       InstallCompletionFilesCommand.commandName,
+      UnistallCompletionFilesCommand.commandName,
     ];
 
+    final completionConfiguration = CompletionConfiguration.fromFile(
+      completionInstallation.completionConfigurationFile,
+    );
+    final isUninstalled = systemShell != null &&
+        completionConfiguration.uninstalls
+            .contains(command: executableName, systemShell: systemShell!);
```

Comment:
This logic should be on `CompletionInstallation`. I is the one that should know if the shell is null or not.

Reply:
Conversation to be continued here: https://github.com/VeryGoodOpenSource/cli_completion/pull/74#discussion_r1202447679

Reply:
Do you think it should return `false` anytime the `systemShell` is null?

Reply:
I refer to every line between 77 and 82. It is installation logic independent of the information stored on `CompletionCommandRunner`. all required information necessary to decide can be found on `CompletionInstallation`.

Reply:
That is true indeed, but wether to exercise the logic or not is dependant if auto-installation in installed (from CompletionCommandRunner). This has been purposely done so as per the reason discussed here https://github.com/VeryGoodOpenSource/cli_completion/pull/74#discussion_r1202447679.

Do you have a more specific suggestion regarding moving the logic to`CompletionInstallation`, like for example, are you suggesting to pass an auto-install flag, new methods, etc? How do you picture the refactor?

Reply:
[Suggestion](https://github.com/VeryGoodOpenSource/cli_completion/pull/74#discussion_r1202716908)

Reply:
I amended the logic let me know what you think. I will update the tests accordingly, since now calling `install` requires to mock the `CompletionConfiguration`.